backend/src/test/java/com/bakdata/conquery/integration/tests/ImportUpdateTest.java

```diff
@@ -0,0 +1,210 @@
+
+            //create import descriptor
+            {
+                TableImportDescriptor desc = new TableImportDescriptor();
+                desc.setName(importTable.getName());
+                desc.setTable(importTable.getName());
+                TableInputDescriptor input = new TableInputDescriptor();
+                {
+                    input.setPrimary(IntegrationUtils.copyOutput(importTable.getPrimaryColumn()));
+                    input.setSourceFile(csvName);
+                    input.setOutput(new OutputDescription[importTable.getColumns().length]);
+                    for (int i = 0; i < importTable.getColumns().length; i++) {
+                        input.getOutput()[i] = IntegrationUtils.copyOutput(importTable.getColumns()[i]);
+                    }
+                }
+                desc.setInputs(new TableInputDescriptor[]{input});
+                Jackson.MAPPER.writeValue(descriptionFile, desc);
+            }
+
+            //preprocess
+            conquery.preprocessTmp(conquery.getTmpDir(), List.of(descriptionFile));
+
+            log.info("updating import");
+            //correct update of the import
+            LoadingUtil.updateCqppFile(conquery, newPreprocessedFile, importId1, Response.Status.Family.SUCCESSFUL, "No Content");
+            conquery.waitUntilWorkDone();
+        }
+
+        // State after update.
+        {
+            log.info("Checking state after update");
+            assertThat(namespace.getStorage().getAllImports()).hasSize(1);
+            // Must contain the import.
+            assertThat(namespace.getStorage().getAllImports())
+                    .filteredOn(imp -> imp.getId().equals(importId1))
+                    .isNotEmpty();
+
+            assertThat(namespace.getStorage().getCentralRegistry().getOptional(importId1))
+                    .isNotEmpty();
+
+            for (ShardNode node : conquery.getShardNodes()) {
+                for (Worker worker : node.getWorkers().getWorkers().values()) {
+                    if (!worker.getInfo().getDataset().equals(dataset.getId())) {
+                        continue;
+                    }
+
+                    final ModificationShieldedWorkerStorage workerStorage = worker.getStorage();
+
+                    assertThat(workerStorage.getAllCBlocks())
+                            .describedAs("CBlocks for Worker %s", worker.getInfo().getId())
+                            .filteredOn(block -> block.getBucket().getId().getDataset().equals(dataset.getId()))
+                            .isNotEmpty();
+
+                    assertThat(workerStorage.getAllBuckets())
+                            .filteredOn(bucket -> bucket.getId().getDataset().equals(dataset.getId()))
+                            .describedAs("Buckets for Worker %s", worker.getInfo().getId())
+                            .isNotEmpty();
```

Comment:
Hier wäre es interessant, ob sich die Bucket auch wirklich verändert haben, oder ob immer noch die alten Buckets drinne sind

Reply:
du hast Recht, ich denke aber, wenn die neuen Entitäten dazu gekommen und zu finden sind, dann können wir daraus schließen, dass die Buckets auch in Ordnung sind. Außerdem testen wir ja letztendlich  das Update der Entitäten

Reply:
Ich fände es hier auch nicht schlecht eine Abfrage zu verschicken, denn das sind dann genau so bugs die dann doch erst beim testen auffallen, dass irgendwas komisch hängt. Kannst mal bei ImportDeletionTest schauen, wie das da gemacht wird. Ist etwas stressig, aber damit sicherst du dich wirklich gegen Lücken in der QueryEngine ab